static/app/views/settings/components/forms/controls/rangeSlider.tsx

```diff
@@ -46,160 +48,159 @@ type Props = {
-    let actualValue = sliderValue;
-    let displayValue: React.ReactNode = actualValue;
-
-    if (allowedValues) {
-      step = 1;
-      min = 0;
-      max = allowedValues.length - 1;
-      actualValue = allowedValues[sliderValue];
-      displayValue =
-        typeof actualValue !== 'undefined' ? actualValue : t('Invalid value');
+  }
+
+  function getSliderData() {
+    if (!allowedValues) {
+      const {min, max, step} = props;
+      return {
+        min,
+        max,
+        step,
+        actualValue: sliderValue,
+        displayValue: sliderValue,
+      };
     }
 
-    displayValue =
-      typeof formatLabel === 'function' ? formatLabel(actualValue) : displayValue;
-
-    return (
-      <div className={className}>
-        {!showCustomInput && <Label htmlFor={name}>{displayValue}</Label>}
-        <SliderAndInputWrapper showCustomInput={showCustomInput}>
-          <Slider
-            type="range"
-            name={name}
-            min={min}
-            max={max}
-            step={step}
-            disabled={disabled}
-            onInput={this.handleInput}
-            onChange={() => {}}
-            onMouseUp={this.handleBlur}
-            onKeyUp={this.handleBlur}
+    const actualValue = allowedValues[sliderValue];
+
+    return {
+      step: 1,
+      min: 0,
+      max: allowedValues.length - 1,
+      actualValue,
+      displayValue: defined(actualValue) ? actualValue : t('Invalid value'),
+    };
+  }
+
+  const {min, max, step, actualValue, displayValue} = getSliderData();
+
+  return (
+    <div className={className} ref={forwardRef}>
+      {!showCustomInput && (
+        <Label htmlFor={name}>
+          {typeof formatLabel === 'function' ? formatLabel(actualValue) : displayValue}
+        </Label>
+      )}
+      <SliderAndInputWrapper showCustomInput={showCustomInput}>
+        <Slider
+          type="range"
+          name={name}
+          min={min}
+          max={max}
+          step={step}
+          disabled={disabled}
+          onInput={handleInput}
+          onMouseUp={handleBlur}
+          onKeyUp={handleBlur}
+          value={sliderValue}
+          hasLabel={!showCustomInput}
+        />
+        {showCustomInput && (
+          <Input
+            placeholder={placeholder}
             value={sliderValue}
-            hasLabel={!showCustomInput}
+            onChange={handleCustomInputChange}
+            onBlur={handleInput}
           />
-          {showCustomInput && (
-            <Input
-              placeholder={placeholder}
-              value={sliderValue}
-              onChange={this.handleCustomInputChange}
-              onBlur={this.handleCustomInputBlur}
-            />
-          )}
-        </SliderAndInputWrapper>
-      </div>
-    );
-  }
+        )}
+      </SliderAndInputWrapper>
+    </div>
+  );
 }
 
-export default RangeSlider;
+const RangeSliderContainer = React.forwardRef(function RangeSliderContainer(
+  props: Props,
+  ref: React.Ref<any>
```

Comment:
Possible to do better than `any` here?

Reply:
it was `HTMLDivElement` before, but I had to update it with any as the tsc was not passing